### PR TITLE
Remove service.configuration and inject directly the service definition configuration to service.dependencies

### PR DESCRIPTION
**Pull request by NicolasMahe**
Solves https://forum.mesg.com/t/service-configuration-is-duplicated-in-service-dependencies/143

### CODE REVIEW

File: service/service.go [hunk 1]
```diff
@@ -268,3 +263,8 @@ func (s *Service) validateConfigurationEnv(env map[string]string) error {
 	}
 	return nil
 }
+
+// helper to return the configuration of the service from the dependencies array
+func (s *Service) configuration() *Dependency {
```
[the remark this answers is not in the document]
Comment (NicolasMahe) [reply service/service.go]: As you say, at least 3 functions need to get the configuration. So I think it's totally valid to extract the logic in an helper function.

File: service/service.go [hunk 2]
```diff
@@ -268,3 +263,8 @@ func (s *Service) validateConfigurationEnv(env map[string]string) error {
...
+	return s.Dependencies[len(s.Dependencies)-1]
```
[the remark this answers is not in the document]
Comment (NicolasMahe) [reply service/service.go]: I don't also but from a logic point of view, the configuration is ALWAYS at the latest index of the Dependencies array. See https://github.com/mesg-foundation/core/pull/667/files#diff-d391c3871baee97866809cf838da79b0R31:
```go
s.Dependencies = append(s.Dependencies, configuration)
```
---
I will implement this based on a constant key `service` and prevent dependencies to use this key in the mesg.yml.

[the remark this answers is not in the document]
Comment (NicolasMahe) [reply service/service.go]: Done

File: service/importer/validation_test.go [hunk 3]
```diff
@@ -78,3 +79,10 @@ func TestIsValidMalFormattedServiceFile(t *testing.T) {
 	_, err := IsValid("./tests/service-file-mal-formatted")
 	require.Error(t, err)
 }
+
+func TestInvalidDependencyName(t *testing.T) {
+	validation, err := Validate("./tests/service-invalid-dependency-name")
+	require.NoError(t, err)
+	require.Len(t, validation.ServiceFileWarnings, 1)
+	fmt.Println("validation.ServiceFileWarnings", validation.ServiceFileWarnings)
```
Comment (ilgooz) [inline service/importer/validation_test.go]:
```suggestion
```

**Pull request merged**